ext/mini_racer_extension/mini_racer_extension.cc

```diff
@@ -981,8 +994,44 @@ static VALUE rb_context_init_unsafe(VALUE self, VALUE isolate, VALUE snap) {
 
         Local<Context> context = Context::New(isolate_info->isolate);
 
+        Context::Scope context_scope { context };
+
         context_info->context = new Persistent<Context>();
         context_info->context->Reset(isolate_info->isolate, context);
+
+        Local<String> strClassName = String::NewFromUtf8Literal(isolate_info->isolate, "RubyError");
+
+        /* Create `RubyError` constructor + prototype (and steal `Error.captureStackTrace` while we're at it)
+         * equivalent JS code is roughly
+         * ```
+         * function RubyError() { throw new Error("This class may not be constructed from JS!") }
+         * RubyError.prototype = Object.create(Error.prototype);
+         * RubyError.prototype.name = "RubyError";
+         * RubyError.prototype.constructor = RubyError;
+         * ```
+         */
+        Local<Object> global = context->Global();
+        Local<Object> errorConstructor = global
+            ->Get(context, String::NewFromUtf8Literal(isolate_info->isolate, "Error"))
+            .ToLocalChecked().As<Object>();
+        Local<Function> captureStackTrace = errorConstructor
+            ->Get(context, String::NewFromUtf8Literal(isolate_info->isolate, "captureStackTrace"))
+            .ToLocalChecked().As<Function>();
+        context_info->capture_stack_trace.Reset(isolate_info->isolate, captureStackTrace);
```

Comment:
can you explain to me using Reset here? As a brand new Local<> there shouldn't be anything to Reset from my understanding.

Reply:
`Global<T>::Reset(Isolate*, Local<T> value)` puts the value of `value` into the `Global<T>` it is called on.

So basically, this is just storing `captureStackTrace` in `context_info->capture_stack_trace`

Reply:
Understood